AmericanExpress/src/main/java/com/braintreepayments/api/americanexpress/AmericanExpressClient.kt

```diff
@@ -0,0 +1,87 @@
+package com.braintreepayments.api.americanexpress
+
+import android.content.Context
+import android.net.Uri
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.core.ApiClient.Companion.versionedPath
+import com.braintreepayments.api.core.BraintreeClient
+import org.json.JSONException
+
+/**
+ * Used to integrate with Braintree's American Express API
+ */
+class AmericanExpressClient {
+    private val braintreeClient: BraintreeClient
```

Comment:
Could we move this property into the primary constructor of the class? This way we could remove this constructor:

```kotlin
    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
    constructor(braintreeClient: BraintreeClient) {
        this.braintreeClient = braintreeClient
    }
```